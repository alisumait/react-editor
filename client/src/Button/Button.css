.save-button,
.delete-button,
.text-button,
.submit-button {
    width: 100%;
}

.text-button,
.submit-button {
    margin-top: 10px;
}
