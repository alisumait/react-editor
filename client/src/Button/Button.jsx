import React from 'react';
import './Button.css';

const Button = ({ children, onClick, disabled, className, ...props }) => {   
  return (
    <button disabled={disabled} className={className} onClick={onClick}>
     {children}    
    </button>
  );
};

export default Button;