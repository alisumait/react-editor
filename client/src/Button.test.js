import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../src/Button/Button';

test('test if onClick method works', () => {
  const component = renderer.create(
    <Button onClick={(e) => console.log("works!")}>Test</Button>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onClick();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});