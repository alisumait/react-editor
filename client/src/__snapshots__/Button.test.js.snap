// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test if onClick method works 1`] = `
<button
  onClick={[Function]}
>
  Test
</button>
`;

exports[`test if onClick method works 2`] = `
<button
  onClick={[Function]}
>
  Test
</button>
`;
